addons/external_intercom/functions/fnc_connect.sqf

```diff
@@ -0,0 +1,148 @@
+if (!isNil {_oldVehicle}) then {
+    [_oldVehicle, _player] call TFAR_external_intercom_fnc_disconnect;
+};
+
+private _vehicleID = _vehicle getVariable ["TFAR_vehicleIDOverride", netId _vehicle];
+_vehicle setVariable ["TFAR_vehicleIDOverride", _vehicleID, true];
+_player setVariable ["TFAR_ExternalIntercomVehicle", _vehicle, true];
+_player setVariable ["TFAR_vehicleIDOverride", _vehicleID, true];
+
+if (_wireless) then {
+    (_externalSpeakers select 1) pushBack _player;
+} else {
+    _externalSpeakers set [0, _player];
+};
+_vehicle setVariable ["TFAR_ExternalIntercomSpeakers", _externalSpeakers, true];
+
+// Get intercom channel, now that we've signed on to the intercom
+private _intercomSlot = [_vehicle] call TFAR_external_intercom_fnc_getIntercomChannel;
+
+_vehicle setVariable [format ["TFAR_IntercomSlot_%1",(netId _player)], _intercomSlot, true];
+_player setVariable [format ["TFAR_IntercomSlot_%1",(netId _player)], _intercomSlot, true]; // Because TFAR checks the "vehicle", and thinks the unit is the vehicle when it's not in a vehicle, so we manually set it
+
+// Show indicator
+if (TFAR_oldVolumeHint) then {
+    [parseText "Connected to Vehicle Intercom", -1] call TFAR_fnc_showHint;
+} else {
+    (QGVAR(PhoneConnectionIndicatorRsc) call BIS_fnc_rscLayer) cutRsc [QGVAR(PhoneConnectionIndicatorRsc), "PLAIN", 0, true];
+};
+
+// Keep connection alive until out of range or disconnected
+private _interactionPointRelative = getCenterOfMass _vehicle;
+private _intercomMaxRange = TFAR_externalIntercomMaxRange_Wireless;
+private _headgear = "";
+
+if (!_wireless) then {
+    _interactionPointRelative = [_vehicle] call TFAR_external_intercom_fnc_getInteractionPoint;
+    _intercomMaxRange = TFAR_externalIntercomMaxRange_Phone;
+} else {
+    _headgear = headgear _player;
+};
+
+[{
+    params ["_vehicle", "_player", "_intercomMaxRange", "_interactionPointRelative", "_wireless", "_headgear"];
+    ((_vehicle modelToWorld _interactionPointRelative) distance _player) > _intercomMaxRange
+    || isNil {_player getVariable "TFAR_ExternalIntercomVehicle"}
+    || (_wireless && !(_headgear isEqualTo (headgear _player)))
+}, {
+    params ["_vehicle", "_player"];
+    [_vehicle, _player] call TFAR_external_intercom_fnc_disconnect;
+}, [_vehicle, _player, _intercomMaxRange, _interactionPointRelative, _wireless, _headgear]] call CBA_fnc_waitUntilAndExecute;
+
+
+if !(_wireless) then {
+    private _handset = createSimpleObject [QPATHTOF(data\TFAR_handset.p3d), _player selectionPosition "head"];
+    _handset attachTo [_player, [-0.14,-0.02,0.02], "head", true];
+    _handset setVectorDirAndUp [[-2.5,0.8,0.25],[-1,-1,1]];
+    private _ropeID = ropeCreate [_vehicle, _interactionPointRelative vectorAdd [0,0.2,0], _handset, "plug", _intercomMaxRange]; // handset rope location[0,0.006,-0.7]
+    _vehicle setVariable ["TFAR_ExternalIntercomRopeIDs", [_ropeID, _handset], true];
+
+    _vehicle addEventHandler ["RopeBreak", {
```

Comment:
Hah. This alone might be a sufficient "out of range" check :D

Reply:
I wish it was. The rope will stretch for ages rather than snap... which is kind of annoying. I wish we could set the elasticity of the rope to tell the engine how much give it should allow.

Actually... Does it exist?

Reply:
Once v2.06 of Arma is released we'll be able to make a custom rope to imitate wire, and change elasticity accordingly.

The code:
```c
class CfgVehicles {
    class Rope;
    class RopeRabbit : Rope {
        maxRelLenght = 1.1;
        maxExtraLenght = 20;
        springFactor = 0.5; // higher == less stretchy rope
        segmentType = "RopeSegmentRabbit";
        torqueFactor = 0.5;
        dampingFactor[] = {1.0,2.5,1.0};
        model = "\A3\Animals_F\rabbit\rabbit_F.p3d";
    };
};

class CfgNonAIVehicles {
    class RopeSegmentRabbit {
        scope=2;
        displayName="";
        simulation="ropesegment";
        autocenter=0;
        animated=0;
        model = "\A3\Animals_F\rabbit\rabbit_F.p3d";
    };
};
```
And to create it, just name the classname as an additional parameter to the usual ones for `ropeCreate`

Reply:
Did you figure out how/when _ropeBreak_ works? Seems to work for direct connect to units for me but as soon as the ropeAttached object is attached itself, it wont. Still investigating if a workaround is possible.

Reply:
Nope, though if what you say is the case... maybe a base-game bug or something? Or maybe the eventhandler needs to be applied to something/one else? I would have thought the vehicle were enough, but maybe it needs to be applied to the phone too?

Reply:
I am working on a phone mod and attach the rope from helperobject (heliclass) to player (lefthand selection) and with this setup ropeBreak works. you need to add the EH to the helperobject (heli class).